AnnService/src/Core/VectorIndex.cpp

```diff
@@ -418,3 +432,41 @@ VectorIndex::LoadIndex(const std::string& p_config, const std::vector<ByteArray>
 
     return p_vectorIndex->LoadIndex(p_config, p_indexBlobs);
 }
+
+
+std::uint64_t VectorIndex::EstimatedVectorCount(std::uint64_t p_memory, DimensionType p_dimension, IndexAlgoType p_algo, VectorValueType p_valuetype, int p_treeNumber, int p_neighborhoodSize)
+{
+    size_t treeNodeSize;
+    if (p_algo == IndexAlgoType::BKT) {
+        treeNodeSize = sizeof(SizeType) * 3;
+    }
+    else if (p_algo == IndexAlgoType::KDT) {
+        treeNodeSize = sizeof(SizeType) * 2 + sizeof(DimensionType) + sizeof(float);
+    }
+    else {
+        return 0;
+    }
+    std::uint64_t unit = GetValueTypeSize(p_valuetype) * p_dimension + sizeof(std::uint64_t) + sizeof(SizeType) * p_neighborhoodSize + 1 + treeNodeSize * p_treeNumber;
+    return p_memory / unit;
+}
+
+
+std::uint64_t VectorIndex::EstimatedMemoryUsage(std::uint64_t p_vectorCount, DimensionType p_dimension, IndexAlgoType p_algo, VectorValueType p_valuetype, int p_treeNumber, int p_neighborhoodSize)
```

Comment:
could you do some experiment to give us some sense of the accuracy of this estimation? like give us a range the actual memory usage like [0,8, 1] of the estimated value.

what's more we need another non-static function to give the memory usage of the sptag instance

Reply:
This calculation only skip some sizeof(int) and dynamic memory usage (it related to the thread number).

We have non-static function to give memory usage. I think I have added in the wrapper.